### PR TITLE
Fix for #101: PoultryInspector unserializability

### LINKED ISSUE #101: PoultryInspector serialization issues
Not entirely sure which component.

    LOG{SEVERE} (SaveLoadComponent.java:146) Could not save!
    java.io.NotSerializableException: ccre.channel.EventOutput$$Lambda$22/968188688
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1184)
        at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1548)
        at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1509)
        at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
        at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
        at java.util.LinkedList.writeObject(LinkedList.java:1131)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1028)
        at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1496)
        at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1432)
        at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1178)
        at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:348)
        at ccre.supercanvas.SuperCanvasPanel.save(SuperCanvasPanel.java:229)
        at ccre.supercanvas.components.pinned.SaveLoadComponent.saveLayout(SaveLoadComponent.java:119)
        at ccre.supercanvas.components.pinned.SaveLoadComponent.onInteract(SaveLoadComponent.java:85)
        at ccre.supercanvas.components.pinned.SaveLoadComponent.onSelect(SaveLoadComponent.java:146)
        at ccre.supercanvas.SuperCanvasPanel$SuperCanvasMouseAdapter.mousePressed(SuperCanvasPanel.java:340)
        at java.awt.Component.processMouseEvent(Component.java:6532)
        at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
        at java.awt.Component.processEvent(Component.java:6300)
        at java.awt.Container.processEvent(Container.java:2236)
        at java.awt.Component.dispatchEventImpl(Component.java:4891)
        at java.awt.Container.dispatchEventImpl(Container.java:2294)
        at java.awt.Component.dispatchEvent(Component.java:4713)
        at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
        at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4522)
        at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
        at java.awt.Container.dispatchEventImpl(Container.java:2280)
        at java.awt.Window.dispatchEventImpl(Window.java:2750)
        at java.awt.Component.dispatchEvent(Component.java:4713)
        at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
        at java.awt.EventQueue.access$500(EventQueue.java:97)
        at java.awt.EventQueue$3.run(EventQueue.java:709)
        at java.awt.EventQueue$3.run(EventQueue.java:703)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
        at java.awt.EventQueue$4.run(EventQueue.java:731)
        at java.awt.EventQueue$4.run(EventQueue.java:729)
        at java.security.AccessController.doPrivileged(Native Method)
        at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
        at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
        at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
        at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
        at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
        at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

### PR DESCRIPTION
**Pull request by col6y**
Fixes #101.

### CODE REVIEW

File: PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        canvas.add(npc = new NetworkPaletteComponent(300, 400));
+        canvas.add(new TopLevelPaletteComponent(400, 300));
+        int x = 0;
+        for (String name : new String[] { "test-boolean", "test-float", "test-event", "test-boolean.input", "test-float.input", "test-event.input", "test-boolean.output", "test-float.output", "test-event.output", "test-logging", "test-output", "test-rconf", "test-rpc" }) {
+            canvas.add(npc.getComponentFor(x, 300, name));
+            x += 50;
+        }
+        File f = File.createTempFile("pstest-", ".ser");
+        f.deleteOnExit();
+        ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(f));
```
Comment (amackworth) [inline PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java]: Perhaps this should use a try-with-resources?

File: PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        try {
+            canvas.save(out);
+        } finally {
+            out.close();
+        }
+        canvas = new SuperCanvasPanel();
+        ObjectInputStream in = new ObjectInputStream(new FileInputStream(f));
```
Comment (amackworth) [inline PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java]: Same here.

**Comment by amackworth** [comment]
Other than a minor quibble, this looks good to go!

File: PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java (hunk 1 above)
Comment (col6y) [inline PoultryInspector/tests/ccre/supercanvas/SerializationIntegrationTest.java]: I have no clue how I missed that.

**Comment by amackworth** [comment]
:+1:

**Pull request merged**